internal/log/retry/retry_test.go

```diff
@@ -25,490 +25,352 @@ import (
-				name: "returns retry object when WithError and WithWarn options is on",
-				opts: []Option{
-					WithWarn(warnFn),
-					WithError(errorFn),
+				name: "returns l when opts is not nil",
+				args: args{
+					opts: []Option{
+						WithError(fn),
+					},
 				},
-				want: &retry{
-					warnFn:  warnFn,
-					errorFn: errorFn,
+				want: want{
+					want: &retry{
+						errorFn: fn,
+						warnFn:  nopFunc,
+					},
 				},
 			}
 		}(),
 	}
 
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			got, ok := New(tt.opts...).(*retry)
-			if !ok {
-				t.Errorf("type is invalid")
+	for _, test := range tests {
+		t.Run(test.name, func(tt *testing.T) {
+			defer goleak.VerifyNone(tt)
+			if test.beforeFunc != nil {
+				test.beforeFunc(test.args)
 			}
-
-			if reflect.ValueOf(got.errorFn).Pointer() != reflect.ValueOf(tt.want.errorFn).Pointer() {
-				t.Error("errorfn is not equals")
+			if test.afterFunc != nil {
+				defer test.afterFunc(test.args)
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
 			}
 
-			if reflect.ValueOf(got.warnFn).Pointer() != reflect.ValueOf(tt.want.warnFn).Pointer() {
-				t.Error("warnfn is not equals")
+			got := New(test.args.opts...)
+			if err := test.checkFunc(test.want, got); err != nil {
+				tt.Errorf("error = %v", err)
 			}
 		})
 	}
 }
 
-func TestOut(t *testing.T) {
+func Test_retry_Out(t *testing.T) {
 	type args struct {
-		fn     func(vals ...interface{}) error
-		format string
-		vals   []interface{}
+		fn   func(vals ...interface{}) error
+		vals []interface{}
 	}
-
-	type field struct {
-		warnFn  func(...interface{})
-		errorFn func(...interface{})
+	type fields struct {
+		warnFn  func(vals ...interface{})
+		errorFn func(vals ...interface{})
 	}
-
 	type test struct {
-		name      string
-		args      args
-		field     field
-		panicked  bool
-		checkFunc func() error
+		name       string
+		args       args
+		fields     fields
+		checkFunc  func() error
+		beforeFunc func(args)
+		afterFunc  func(args, *testing.T)
+	}
+	defaultCheckFunc := func() error {
+		return nil
 	}
-
 	tests := []test{
 		func() test {
-			cnt := 0
+			var (
+				wantCnt = 1
+				gotCnt  = 0
+			)
+
 			fn := func(vals ...interface{}) error {
-				cnt++
+				gotCnt++
 				return nil
 			}
+
 			return test{
-				name: "returns nothing when fn returns nil",
+				name: "called success when fn returns nil",
 				args: args{
 					fn: fn,
 				},
 				checkFunc: func() error {
-					if cnt != 1 {
-						return errors.Errorf("called cnt is wrong. want: %v, but got: %v", 1, cnt)
+					if gotCnt != wantCnt {
+						return errors.Errorf("count: got: %d, want: %d", gotCnt, wantCnt)
 					}
 					return nil
 				},
 			}
 		}(),
 
-		func() test {
-			return test{
-				name: "returns nothing when fn is nil",
-				checkFunc: func() error {
-					return nil
-				},
-			}
-		}(),
+		{
+			name: "returns nothing when fn is nil",
+			checkFunc: func() error {
+				return nil
```

Comment:
it doesn't check anything?

Reply:
thank you for your review.
This test case is not nessesary. I fixed the early return in the implementation code, so I removed this case.

Reply:
ok!